python/ray/dataframe/concat.py

```diff
@@ -21,45 +21,38 @@ def _concat(frame1, frame2):
                              ignore_index, keys, levels, names,
                              verify_integrity, copy)
 
-        if not (isinstance(frame1, rdf) and
-           isinstance(frame2, rdf)) and join == 'inner':
-            raise NotImplementedError(
-                  "Obj as dicts not implemented. To contribute to "
-                  "Pandas on Ray, please visit github.com/ray-project/ray."
-                  )
-
+        f1_columns, f2_columns = frame1.columns, frame2.columns
+        
         # Case 2: Both are different types
         if isinstance(frame1, pd.DataFrame):
             frame1 = from_pandas(frame1, len(frame1) / 2**16 + 1)
+            frame1.columns = f1_columns
```

Comment:
from_pandas already sets the columns in the output df

Reply:
They aren't being set properly though when converting a pd.DataFrame in the mixed frame tests. Use the df, df2, df3 test and put a print after to verify. This is where the `0, 1, 2, 4` column names are coming from

Reply:
Resolved